Server/ReasnAPI/ReasnAPI/Services/CommentService.cs

```diff
@@ -1,16 +1,103 @@
-﻿using ReasnAPI.Models.Database;
-
-namespace ReasnAPI.Services {
-    public class CommentService (ReasnContext context) {
-        private readonly ReasnContext _context = context;
-
-        /* TODO: Create following functions for this class
-         * create
-         * update
-         * delete
-         * get by ID
-         * get list by filter
-         * get all
-         */
-    }
-}
+using ReasnAPI.Mappers;
+using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using ReasnAPI.Services.Exceptions;
+using System.Linq.Expressions;
+
+namespace ReasnAPI.Services;
+
+public class CommentService(ReasnContext context)
+{
+    public CommentDto? CreateComment(CommentDto? commentDto)
+    {
+        ArgumentNullException.ThrowIfNull(commentDto);
+
+        context.Comments.Add(commentDto.ToEntity());
+        context.SaveChanges();
+
+        return commentDto;
+    }
+
+    public CommentDto? UpdateComment(int commentId, CommentDto? commentDto)
+    {
+        ArgumentNullException.ThrowIfNull(commentDto);
+
+        var comment = context.Comments.FirstOrDefault(r => r.Id == commentId);
+
+        if (comment is null)
+        {
+            throw new NotFoundException("Comment not found");
+        }
+
+        var user = context.Users.FirstOrDefault(r => r.Id == commentDto.UserId);
+
+        if (user is null)
+        {
+            throw new NotFoundException("User not found");
+        }
+
+        if (comment.UserId != commentDto.UserId && user.Role != Models.Enums.UserRole.Admin)
+        {
+            throw new BadRequestException("User is not authorized to update this comment");
+        }

```

Comment:
Why do you throw `BadRequestException` when user does not have permission to perform operation - shouldn't it be sth like: `NotAuthorizedException`? Anyway the authorization logic should be perfomed on controller level - the service should be focused on business logic and data manipulation.